[libwebrtc] Start building sdp_integration_fuzzer again

JonWBedard opened the pull request:
#### f2fede649acc6296609f7f9048414e3c091c7538
<pre>
[libwebrtc] Start building sdp_integration_fuzzer again
<a href="https://bugs.webkit.org/show_bug.cgi?id=287154">https://bugs.webkit.org/show_bug.cgi?id=287154</a>
<a href="https://rdar.apple.com/144309158">rdar://144309158</a>

Reviewed by Elliott Williams.

* Source/ThirdParty/libwebrtc/Configurations/sdp_integration_fuzzer.xcconfig: Configure LIBRARY_SEARCH_PATHS
to include the built SDK. Remove gmock and gtest linkages in favor of ones in the xcodeproj.
* Source/ThirdParty/libwebrtc/libwebrtc.xcodeproj/project.pbxproj: Re add sdp_integration_fuzzer, add
gmock and gtest dependencies.

Canonical link: <a href="https://commits.webkit.org/292204@main">https://commits.webkit.org/292204@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c962ffe31a84a2d3d66931c64c8b860f99aeed2c

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/95285 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/14885 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/4743 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/100329 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/45786 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/15173 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/23316 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/72658 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/29934 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/98288 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/11328 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/86005 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/52990 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/11042 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/3743 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/45125 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/81217 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/3838 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/102367 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/22333 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/16285 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/81655 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/22581 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/82025 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/81053 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/20266 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/25623 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/3022 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/15586 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/22303 "Built successfully") | | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/116596 "Build is in progress. Recent messages:") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/21962 "Built successfully") | | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/35/builds/116596 "Build is in progress. Recent messages:") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/25436 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/23701 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

emw-apple commented on Source/ThirdParty/libwebrtc/Configurations/sdp_integration_fuzzer.xcconfig:
```diff
@@ -36,8 +36,7 @@ ALTERNATE_HEADER_SEARCH_PATHS = $(ALTERNATE_HEADER_SEARCH_PATHS_$(SDK_VARIANT));
 ALTERNATE_HEADER_SEARCH_PATHS_iosmac = $(BUILT_PRODUCTS_DIR)$(WK_ALTERNATE_FRAMEWORKS_DIR)$(WK_LIBRARY_HEADERS_FOLDER_PATH);
 
 HEADER_SEARCH_PATHS = $(inherited) $(ALTERNATE_HEADER_SEARCH_PATHS) $(BUILT_PRODUCTS_DIR)$(WK_LIBRARY_HEADERS_FOLDER_PATH);
+LIBRARY_SEARCH_PATHS = $(SDK_DIR)$(WK_LIBRARY_INSTALL_PATH) $(inherited);
```
Putting the SDK path first seems concerning for at-desk builds, because you're telling the linker to prefer libraries from the SDK over anything else.
```suggestion
LIBRARY_SEARCH_PATHS = $(inherited) $(SDK_DIR)$(WK_LIBRARY_INSTALL_PATH);
```

You probably copied this line from JavaScriptCore's config, but I think it's incorrect there too :\

JonWBedard committed:
[libwebrtc] Start building sdp_integration_fuzzer again
https://bugs.webkit.org/show_bug.cgi?id=287154
rdar://144309158

Reviewed by Elliott Williams.

* Source/ThirdParty/libwebrtc/Configurations/sdp_integration_fuzzer.xcconfig: Configure LIBRARY_SEARCH_PATHS
to include the built SDK. Remove gmock and gtest linkages in favor of ones in the xcodeproj.
* Source/ThirdParty/libwebrtc/libwebrtc.xcodeproj/project.pbxproj: Re add sdp_integration_fuzzer, add
gmock and gtest dependencies.

Canonical link: https://commits.webkit.org/292204@main
